cli/crates/cli/tests/auth.rs

```diff
@@ -0,0 +1,95 @@
+mod utils;
+
+use serde_json::{json, Value};
+use utils::environment::Environment;
+
+const ISSUER_URL: &str = "https://some.issuer.test";
+const JWT_SECRET: &str = "topsecret";
+
+#[test]
+fn jwt_provider() {
+    const GROUP: &str = "backend";
+    const SCHEMA: &str = const_format::formatcp!(
+        r#"
+            schema @auth(
+                providers: [ {{ type: jwt, issuer: "{ISSUER_URL}", secret: "{JWT_SECRET}" }} ]
+                rules: [ {{ allow: groups, groups: ["{GROUP}"] }} ]
+            ){{
+                query: Query
+            }}
+
+            type Todo @model {{
+                id: ID!
+                title: String!
+            }}
+        "#
+    );
+    const QUERY: &str = r#"
+        query {
+            todoCollection(first: 1) { __typename }
+        }
+    "#;
+
+    let mut env = Environment::init(4015);
+    env.grafbase_init();
+    env.write_schema(SCHEMA);
+    env.grafbase_dev();
+
+    let client = env.create_client();
+    client.poll_endpoint(30, 300);
+
+    // No auth header -> no authorization done in CLI
+    let resp = client.gql::<Value>(json!({ "query": QUERY }).to_string());
+    let errors: Option<Value> = dot_get_opt!(resp, "errors");
+    assert!(errors.is_none(), "errors: {errors:#?}");
```

Comment:
Those error comparisons look like common operations to me that should be shared between tests. Adding something to utils maybe?

Reply:
We could yes